# **AMERICAN INTERNATIONAL UNIVERSITY-BANGLADESH**
---
## **PROGRAMMING IN PYTHON**
**Section:** C | **Semester:** Spring 25-26
### *Final Term Project*

> #### **Uncovering the Hidden Drivers of Student Depression**
> *An Exploratory Data Analysis of Lifestyle, Academic Pressure and Financial Stress*

**Project Members:**
| Name | Student ID | GitHub Profile |
| :-------- | :----------: | :----------: |
| Salman Sayeed | 22-49006-3 | [salman-sayeed](https://github.com/salman-sayeed) |
| Kayjer Islam | 22-49005-3 | [kayjer-islam](https://github.com/Kayjer-Islam) |

<br> <br>

## A. Problem Defination

#### A.1. Dataset Selection
- For this project, I have selected the [Student Depression Dataset](https://www.kaggle.com/datasets/hopesb/student-depression-dataset) sourced from Kaggle. This dataset contains 27,901 records and 18 columns, covering a diverse range of demographic, academic and lifestyle variables.

#### A.2. Problem Statement
- Student mental health is a growing global concern. This dataset represents the intersection of personal lifestyle choices (sleep, diet), academic environment (pressure, satisfaction), and external socio-economic factors (financial stress). By analyzing this data, we aim to move beyond anecdotal evidence and identify the specific pressures that correlate most strongly with depression in a student population.

#### A.3. Analytical Questions
1.  Which factors are most strongly associated with student depression?
> Student depression is influenced by many academic, financial and personal factors. Identifying which variables show the strongest association helps highlight the most important risk indicators. This can guide universities, families and policymakers toward targeted interventions and support systems.
2. How do lifestyle habits differ between depressed and non-depressed students?
> Lifestyle behaviors such as sleep, diet, work/study hours and satisfaction levels are closely linked to mental health. Understanding how these habits differ between groups can reveal patterns that may contribute to poor mental wellbeing and help promote healthier daily routines for students.
3. Does academic and financial stress significantly increase depression risk?
> Academic pressure and financial stress are common challenges faced by students worldwide. Investigating their relationship with depression can provide evidence of how stress impacts mental health and emphasize the importance of stress management and financial support programs.

<br>

## B. Data Understanding

In [37]:
#All imports will be heree
import os
from google.colab import userdata
import pandas as pd
import numpy as np


In [13]:
#token dhorbina kayjer
try:
    os.environ['KAGGLE_API_TOKEN'] = userdata.get('KAGGLE_TOKEN')
    print("Kaggle token loaded successfully")
except:
    print("Token not found")

Kaggle token loaded successfully


In [19]:
!kaggle datasets download -d hopesb/student-depression-dataset --unzip

#data frame
df = pd.read_csv('Student Depression Dataset.csv')


Dataset URL: https://www.kaggle.com/datasets/hopesb/student-depression-dataset
License(s): apache-2.0
100% 454k/454k [00:00<00:00, 104MB/s]



#### B.1. Shape, columns and data types

In [20]:
print(f"Dataset Shape: {df.shape}")

Dataset Shape: (27901, 18)


> The dataset contains 27,901 rows and 18 columns. The large number of records ensures that our statistical analysis and subgroup comparisons will be robust and less prone to being skewed by a few individual data points.

In [21]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27901 entries, 0 to 27900
Data columns (total 18 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     27901 non-null  int64  
 1   Gender                                 27901 non-null  object 
 2   Age                                    27901 non-null  float64
 3   City                                   27901 non-null  object 
 4   Profession                             27901 non-null  object 
 5   Academic Pressure                      27901 non-null  float64
 6   Work Pressure                          27901 non-null  float64
 7   CGPA                                   27901 non-null  float64
 8   Study Satisfaction                     27901 non-null  float64
 9   Job Satisfaction                       27901 non-null  float64
 10  Sleep Duration                         27901 non-null  object 
 11  Di

> The dataset is a mix of numerical (float64, int64) and categorical (object) variables.
>
>Columns like Age, Academic Pressure, and Work/Study Hours are numeric values and allow for statistical calculations such as means and standard deviations.
>
>Columns like Gender, Sleep Duration, and Dietary Habits are catagorical values and will require encoding or grouping for analysis
>
>Depression is an integer (0 or 1), making it suitable for binary comparison between groups. This will be target variable



#### B.2. Missing Values

In [22]:
print(df.isnull().sum())

id                                       0
Gender                                   0
Age                                      0
City                                     0
Profession                               0
Academic Pressure                        0
Work Pressure                            0
CGPA                                     0
Study Satisfaction                       0
Job Satisfaction                         0
Sleep Duration                           0
Dietary Habits                           0
Degree                                   0
Have you ever had suicidal thoughts ?    0
Work/Study Hours                         0
Financial Stress                         3
Family History of Mental Illness         0
Depression                               0
dtype: int64


> Most columns are complete, but Financial Stress has 3 missing values. While this is a very small percentage of the total data

#### B.3. Summary

In [24]:
df.describe()

,id,Age,Academic Pressure,Work Pressure,CGPA,Study Satisfaction,Job Satisfaction,Work/Study Hours,Financial Stress,Depression
count,27901.000000,27901.000000,27901.000000,27901.000000,27901.000000,27901.000000,27901.000000,27901.000000,27898.000000,27901.000000
mean,70442.149421,25.822300,3.141214,0.000430,7.656104,2.943837,0.000681,7.156984,3.139867,0.585499
std,40641.175216,4.905687,1.381465,0.043992,1.470707,1.361148,0.044394,3.707642,1.437347,0.492645
min,2.000000,18.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000
25%,35039.000000,21.000000,2.000000,0.000000,6.290000,2.000000,0.000000,4.000000,2.000000,0.000000
50%,70684.000000,25.000000,3.000000,0.000000,7.770000,3.000000,0.000000,8.000000,3.000000,1.000000
75%,105818.000000,30.000000,4.000000,0.000000,8.920000,4.000000,0.000000,10.000000,4.000000,1.000000
max,140699.000000,59.000000,5.000000,5.000000,10.000000,5.000000,4.000000,12.000000,5.000000,1.000000


- The Age column (min: 18, max: 59) confirms this dataset includes both traditional students and working professionals or mature students.
- A mean of approximately 0.585 indicates that **58.55%** of the students in this dataset are classified as having depression.
- This shows a relatively balanced distribution, though it leans slightly toward a majority "depressed" class. This is ideal for our analysis because we have a large enough sample size in both groups (depressed vs. non-depressed) to make statistically significant comparisons.

<br>

## C. Daya Cleaning & Preparation

In [30]:
df_clean = df.copy()

print(f"Dataset Shape: {df_clean.shape}")

Dataset Shape: (27901, 18)


#### C.1. Handling Missing Values

In [33]:
df_clean = df_clean.dropna(subset=['Financial Stress'])

print(df_clean.isnull().sum())

print(f"\nDataset Shape: {df_clean.shape}")

id                                       0
Gender                                   0
Age                                      0
City                                     0
Profession                               0
Academic Pressure                        0
Work Pressure                            0
CGPA                                     0
Study Satisfaction                       0
Job Satisfaction                         0
Sleep Duration                           0
Dietary Habits                           0
Degree                                   0
Have you ever had suicidal thoughts ?    0
Work/Study Hours                         0
Financial Stress                         0
Family History of Mental Illness         0
Depression                               0
dtype: int64

Dataset Shape: (27898, 18)


> The missing data represents only 0.01% of the total dataset (3 out of 27,901). By removing these few rows, we ensure that all subsequent calculations, especially the custom Stress Index—are based entirely on original, authentic student responses without needing to "guess" or impute values. This maintains high data integrity for the analysis.

#### C.2. Data Type Conversion (Sleep Duration)

In [34]:
sleep_mapping = {
    '5-6 hours': 5.5,
    '7-8 hours': 7.5,
    'More than 8 hours': 9.0,
    'Less than 5 hours': 4.0
}
df_clean['Sleep Duration'] = df_clean['Sleep Duration'].map(sleep_mapping)

> The Sleep Duration column is currently an "object" (text) type. You cannot calculate a correlation or a mean for a string like "5-6 hours".  Converting this to a numerical format is essential for analysis. This allows us to use NumPy for statistical comparisons and Matplotlib to create boxplots that show the actual distribution of sleep hours among students.

#### C.3. Standardizing Column Names

In [36]:
df_clean.columns = [
    'id', 'gender', 'age', 'city', 'profession', 'academic_pressure',
    'work_pressure', 'cgpa', 'study_satisfaction', 'job_satisfaction',
    'sleep_duration', 'dietary_habits', 'degree', 'suicidal_thoughts',
    'work_study_hours', 'financial_stress', 'family_history', 'depression'
]

> The original column names include spaces, special characters (ex: ?) and inconsistent capitalization. This makes the code harder to write and more prone to syntax errors.
>
> This step improves professionalism and code readability, which is a requirement for a high score in the "Code Quality" section of the marking scheme. It also simplifies the process of calling columns during the visualization and analysis phases.

#### C.4. Standardizing Column Names

In [39]:
# z score
age_mean = np.mean(df_clean['age'])
age_std = np.std(df['age'])
df_clean['age_zscore'] = (df_clean['age'] - age_mean) / age_std

outliers = df_clean[df_clean['age_zscore'].abs() > 3]

print(f"Number of outliers in Age: {len(outliers)}")

Number of outliers in Age: 19


> The Age column may contain extreme values (outliers) that could skew the analysis of "typical" student experiences.
>
> We used NumPy to calculate Z-scores and identify rows that are more than 3 standard deviations from the mean.


<br>

## D. Feature Engineering

#### D.1. Stress Index

In [41]:
stress_columns = ['academic_pressure', 'work_pressure', 'financial_stress']

df_clean['stress_index'] = np.mean(df_clean[stress_columns], axis=1)

> A student might have low academic pressure but high financial stress. The stress_index gives us a "whole-picture" view of the student's burden which is essential for answering Q3.

#### D.2. Lifestyle Score

In [42]:
df_clean['lifestyle_score'] = (df_clean['sleep_duration'] + df_clean['study_satisfaction']) / 2

> This helps us answer Q2 by creating a single metric for "lifestyle health" rather than looking at sleep and satisfaction in isolation.

#### D.3. Age Grouping

In [44]:
bins = [18, 22, 27, 60]
labels = ['Teen (18-22)', 'Young Adult (23-27)', 'Adult (28+)']

df_clean['age_group'] = pd.cut(df_clean['age'], bins=bins, labels=labels, include_lowest=True)

> This allows for subgroup comparisons (e.g., Is depression more prevalent in Teens or Adults?)

In [45]:
print(df_clean.columns)

Index(['id', 'gender', 'age', 'city', 'profession', 'academic_pressure',
       'work_pressure', 'cgpa', 'study_satisfaction', 'job_satisfaction',
       'sleep_duration', 'dietary_habits', 'degree', 'suicidal_thoughts',
       'work_study_hours', 'financial_stress', 'family_history', 'depression',
       'age_zscore', 'stress_index', 'lifestyle_score', 'age_group'],
      dtype='object')


<br>

## E. Data Analysis

#### E.1